geode-assembly/src/distributedTest/java/org/apache/geode/session/tests/Jetty9CachingClientServerTest.java

```diff
@@ -44,29 +45,37 @@ public ContainerInstall getInstall(IntSupplier portSupplier)
    * request
    */
   @Test
-  public void shouldCacheSessionOnClient()
+  public void shouldUpdateSessionAttributesFromServer()
       throws Exception {
     manager.startAllInactiveContainers();
     await().until(() -> {
       ServerContainer container = manager.getContainer(0);
       return container.getState().isStarted();
     });
     String key = "value_testSessionExpiration";
-    String value = "Foo";
+    String localValue = "bogus";
+    String remoteValue = "Foo";
 
     client.setPort(Integer.parseInt(manager.getContainerPort(0)));
-    Client.Response resp = client.set(key, value);
+    Client.Response resp = client.set(key, localValue);
     assertThat(resp.getResponse()).isEqualTo("");
     assertThat(resp.getSessionCookie()).isNotEqualTo("");
 
+    // Make sure the client correctly set its original cached value
+    resp = client.get(key);
+    assertThat(resp.getResponse()).isEqualTo(localValue);
+
     serverVM.invoke("set bogus session key", () -> {
```

Comment:
Probably should change the runnable name here as the test expectation has been changed.

Reply:
Good idea! I've changed it to 'remote session key' instead.